independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/resource/basic/RequestFilterQueryParamsTest.java

```diff
@@ -56,8 +73,9 @@ public static class ReplacePathAndQueryParamsFilter implements ContainerRequestF
         @Override
         public void filter(ContainerRequestContext requestContext)
                 throws IOException {
-            UriBuilder builder = requestContext.getUriInfo().getRequestUriBuilder();
-            builder.replacePath("/hello");
+            UriInfo originalUriInfo = requestContext.getUriInfo();
+            UriBuilder builder = originalUriInfo.getRequestUriBuilder();
+            builder.replacePath(originalUriInfo.getPath().contains("2") ? "/hello/2" : "/hello");
```

Comment:
OH OK I see. I was thinking it's `replacePath` that we need to check, thinking this is where the original reporter added a path that included query parameters that we dropped.

But in fact we're using the builder to build a URI, which includes query parameters, and then later down we call `setRequestUri` with query param (that was the original bug).

I got confused, this is fine, thanks :)